src/starknet.rs

```diff
@@ -584,6 +542,10 @@ pub(crate) mod handler {
         nonce: Felt252Abi,
     }
 
+    /// A C ABI Wrapper around the StarknetSyscallHandler
+    ///
+    /// It contains pointers to functions which can be called through MLIR based on the field offset
+    /// The functions convert C ABI structures to the Rust equivalent and calls the wrapped implementation.
```

Comment:
Thanks for the docs. Since I've been asked that before, could you add that the reason for having to have the function pointers instead of it being like the runtime library is that the vtable has a generic type?